### PR TITLE
Refactor all-questions route

### PR DESCRIPTION
**Pull request by sebruiz**
Using custom hooks to clean up async network interactions. The challenge presented in AllQuestionsByState.tsx is as follows:
We want to lazy load IPFS data after we load chain data, because IPFS data may take longer. We can't pass down the responsibility of loading that data down to the bottom-most component because we want to be able to sort/filter by attributes on the IPFS data.

Solution:
Use a custom hook. Using `useReducer` can help with performing async state updates on the chain data.

### CODE REVIEW

File: app/hooks/useContracts.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    xMetricJson,
+    questionAPIJson,
+    questionStateController,
+    bountyQuestionJson,
+    costController,
+    vaultJson,
+  };
+}
+
+function getContracts(chainId?: Chain["id"]) {
```
Comment (jdaly13) [inline app/hooks/useContracts.ts:25]: I still feel this shouldn't be a front end call and storing all that data unnecessarily we could easily load this data (abi and and addresses) per route create context on top level and do useContext instead of getContracts.  We can discuss more and get others opinions on it, but a I created a POC for that.  I feel like this call needs to be more top level and not be part of a hook, I mentioned this in Mason's PR as well

File: app/hooks/questions.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+}[];
+
+export function useQuestions() {
+  const bountyQuestionContract = useBountyQuestionContract();
+  const questionStateControllerContract = useQuestionStateControllerContract();
+
+  const [questions, setQuestions] = useState<getQuestionsByStateResult>();
+
+  // Get chain data
+  useEffect(() => {
```
Comment (jdaly13) [inline app/hooks/questions.ts:23]: This looks fine - so are we not using wagmi useContractRead hooks any longer but creating our own and just instantiating the contract first?

Comment (jdaly13) [reply app/hooks/questions.ts:23]: It's just a different approach as I thought Wagmi didn't encourage that but instad to use useContractRead and pass that data down to a new component that uses that data - this works too just different architecture - This might make it possibly more scalable but also don't want to overcomplicate things and keep things as stupid as possible.

Comment (sebruiz) [reply app/hooks/questions.ts:23]: The awkward part there is chaining those 2 `useContractRead` calls together. I think you have to use `enable` to disable one of the calls and then use `refetch` in a useEffect to chain them together. I thought this was a little nicer than that. Especially if in the future we do more chaining for reads.

Comment (sebruiz) [reply app/hooks/questions.ts:23]: but the nice part is we can use this hook either way. The interface should remain the same. Only the internals change.

File: app/hooks/useContracts.ts (hunk 1 above)
Comment (sebruiz) [reply app/hooks/useContracts.ts]: This is another scenario where we can keep the hook but change the internals to do a server `fetch`

File: app/components/NetworkRender.tsx [hunk 3]
```diff
@@ -1,52 +1,31 @@
...
+  // TODO: Probably don't need this if/else
+  if (typeof switchNetwork === "function") {
+    return (
+      <div className={`tw-flex tw-flex-wrap tw-max-w-xs tw-text-sm tw-mx-auto`}>
+        <p className="tw-mb-2"> You are currently connected to {chainName}</p>
+        <button
+          onClick={() => switchNetwork()}
+          className="tw-bg-[#21C5F2] tw-px-5 tw-py-3 tw-text-sm tw-rounded-lg tw-text-white"
+        >
+          Switch to {chain?.id === 1 ? "Polygon" : "Ethereum Mainnet"}
```
Comment (jdaly13) [inline app/components/NetworkRender.tsx:19]: I'm sort of confused by this - currently we don't support Ethereum Mainnet so if you're connected to Polygon and that's the chosen network based on the env variable you shouldn't be display a switch button

File: app/hooks/contracts.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    xMetricJson,
+    questionAPIJson,
+    questionStateController,
+    bountyQuestionJson,
+    costController,
+    vaultJson,
+  };
+}
+
+function getContracts(chainId?: Chain["id"]) {
```
Comment (jdaly13) [inline app/hooks/contracts.ts:28]: As of the current implementation this doesn't support working locally and using other chains - it's not backwards compatible -  require was used because you can't conditionally import these files (I believe there is a way but I think it makes more sense to use backend) - I'm out a lot of next week but feel like we should try to set up a time to discuss with the group how we want to handle this because it's a big part of the app architecture - the custom hooks in questions.ts I think should work - I would really prefer to avoid the pattern in this file between using imports and loading all the contents on the front end.
Here's a gist of what I envision - https://gist.github.com/jdaly13/a53ebed99926dfd5bb4bfe0aaeb77a16
Basically we get the contracts on the route level and filter them out on the backend so the front end only gets what it needs the abi and the address and any child component can just implement useContext
But I'm certainly open to other alternatives.

**Comment by thehannahlerman** [comment]
I'm liking the organization of this better
